### PR TITLE
updated to only run on releases

### PR DESCRIPTION
**Pull request by shershah010**
The current deployment workflow will build and update the pypi package on every push to main. This changes it so that the package is updated on ever release instead,

### CODE REVIEW

**Comment by shershah010** [comment]
@tfukaza Just to clarify, once you approve a pr are we free to merge it or will you do that because I know in some repos that person who reviewed it also merges it. Thanks!

**Review by tfukaza — APPROVED** [review]

**Comment by tfukaza** [comment]
@shershah010 I'll leave it to whoever made the PR, in case they want to make some last-minute changes.

**Pull request merged**
